pg-extend/src/pg_fdw.rs

```diff
@@ -216,18 +211,53 @@ impl<T: ForeignData> ForeignWrapper<T> {
         }
     }
 
-    unsafe fn get_table_name(rel: &pg_sys::RelationData) -> String {
+    unsafe fn get_table_meta(rel: &pg_sys::RelationData) -> (OptionMap, OptionMap, String) {
```

Comment:
I agree, I'm changing it now.

What do you think about changing `ForeignData::begin()` and `ForeignData::index_columns()` to receive a reference to the new type instead of owned values of the options & table name? This way we could extract the metadata only once, store it in an internal `Option` and pass the same extracted value as a reference

Reply:
Yeah, I think that's a good idea.